src/hooks/useElementWidth.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+
+const useElementWidth = (
+  ref: React.RefObject<HTMLElement | HTMLButtonElement | HTMLInputElement>
```

Comment:
HTMLButtonElement and HTMLInputElement is wider type of HTMLElement

```suggestion
  ref: React.RefObject<HTMLElement>
```